src/utils.ts

```diff
@@ -1961,15 +2042,19 @@ const TOOLTIP_TEXT_OFFSET = 10;
 const TOOLTIP_TRIANGLE_SIZE = 10;
 export class newTooltip {
   public path: Path2D;
+
   public strokeStyle: string = "hsl(210, 90%, 20%)";
   public textColor: string = "hsl(0, 0%, 100%)";
   public fillStyle: string = "hsl(210, 90%, 20%)";
   public alpha: number = 0.8;
-  public fontsize: number = 12;
+  public fontsize: number = 10;
   public radius: number = 10;
+
   private printedRows: string[];
-  private size: Vertex;
   private squareOrigin: Vertex;
+  private size: Vertex;
+  private isUp = true;
+  public isFlipper = false;
```

Comment:
What does isFlipper denote ?

Reply:
It is here to say if tooltip has to go down or to flip when hitting the top of drawing zone